crates/core/src/surfnet/mod.rs

```diff
@@ -327,6 +365,10 @@ impl SurfnetSvm {
         self.inner.set_sysvar(
             &solana_sdk::sysvar::recent_blockhashes::RecentBlockhashes::from_iter(entries),
         );
+        BlockIdentifier::new(
+            self.chain_tip.index + 1,
+            latest_entry.blockhash.to_string().as_str(),
```

Comment:
Would storing a `Hash` for our block identifier hash prevent unnecessary conversions? I think `Hash` implements Copy as well, so it's pretty handy

Reply:
At some point we'll need to render this value as a string. If we convert a hash into a string here, we do this operation once. If we keep a hash, we'll recompute the representation every time it's being rendered?